client/layout/masterbar/quick-language-switcher.jsx

```diff
@@ -0,0 +1,59 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { getLocaleSlug } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+
+import MasterbarItem from './item';
+import LanguagePickerModal from 'components/language-picker/modal';
+import switchLocale from 'lib/i18n-utils/switch-locale';
+import config from 'config';
+
+class QuickLanguageSwitcher extends React.Component {
+	state = {
+		isShowingModal: false,
+	};
+
+	toggleLanguagesModal = () => {
+		this.setState( state => ( {
+			isShowingModal: ! state.isShowingModal,
+		} ) );
+	};
+
+	onClick = event => {
+		this.toggleLanguagesModal();
+		event.preventDefault();
+		return;
+	};
+
+	onSelected = languageSlug => {
+		switchLocale( languageSlug );
+	};
+
+	render() {
+		const selectedLanguageSlug = getLocaleSlug();
+
+		return (
+			<div className="masterbar__item quick-language-switcher">
+				<MasterbarItem icon="globe" onClick={ this.onClick }>
```

Comment:
A pedantic point: because we're not passing a `url` prop, the cursor displays as default.

We could show a pointer, indicating a live hotspot, by adding the following to `client/layout/masterbar/style.scss`:

```
.masterbar__quick-language-switcher a {
	cursor: pointer;
}
```

Result:

![aug-30-2018 11-14-28](https://user-images.githubusercontent.com/6458278/44823830-e5a02a00-ac45-11e8-8959-04ebe06fb45f.gif)